src/common/ant-registry.ts

```diff
@@ -70,6 +79,41 @@ export class AoANTRegistryReadable implements AoANTRegistryRead {
     } else {
       throw new InvalidContractConfigurationError();
     }
+
+    if (config?.hyperbeamUrl !== undefined) {
+      this.hyperbeamUrl = new URL(config.hyperbeamUrl).toString();
+      this.checkHyperBeamPromise = this.checkHyperBeamCompatibility();
+    }
+  }
+
+  /**
+   * Check if the process is HyperBeam compatible. If so, we'll use the HyperBeam node to fetch the state.
+   *
+   * @returns {Promise<boolean>} True if the process is HyperBeam compatible, false otherwise.
+   */
+  private async checkHyperBeamCompatibility(): Promise<boolean> {
```

Comment:
not: put in utility and use it various places, including ANTs

Reply:
This feels a bit funky because its a util thats modifying a property on the class (checkHyperBeamPromise)

How about we start a HyperBEAM class that stores the promise there, can be built with the logger, url, processId, and its just this method for the moment? Then the classes like ANT and ANTRegistry can be built with that in the same concept of the AOProcess class.

Later we can add read api to it as well, eg for compute vs cache with pathing and device configuration.

Thoughts?